api/src/main/java/org/openmrs/module/patientqueueing/api/dao/PatientQueueingDao.java

```diff
@@ -28,4 +38,56 @@ public DbSession getSession() {
 		return sessionFactory.getCurrentSession();
 	}
 	
+	/**
+	 * @see org.openmrs.module.patientqueueing.api.PatientQueueingService#getPatientQueueById(Integer)
+	 */
+	public PatientQueue getPatientQueueById(Integer queueId) {
+		return (PatientQueue) getSession().createCriteria(PatientQueue.class)
+		        .add(Restrictions.eq("patientQueueId", queueId)).uniqueResult();
+	}
+	
+	/**
+	 * @see org.openmrs.module.patientqueueing.api.PatientQueueingService#getPatientQueueList(Provider,
+	 *      Date, Date, Location, Location, Patient, String)
+	 */
+	public List<PatientQueue> getPatientQueueList(Provider provider, Date fromDate, Date toDate, Location locationTo,
+	        Location locationFrom, Patient patient, String status) {
+		Criteria criteria = getSession().createCriteria(PatientQueue.class);
+		if (fromDate != null && toDate != null) {
+			criteria.add(Restrictions.between("dateCreated", fromDate, toDate));
+		}
+		
+		if (provider != null) {
+			criteria.add(Restrictions.eq("provider", provider));
+		}
+		
+		if (locationTo != null) {
+			criteria.add(Restrictions.eq("locationTo", locationTo));
+		}
+		
+		if (locationFrom != null) {
+			criteria.add(Restrictions.eq("locationFrom", locationFrom));
+		}
+		
+		if (patient != null) {
+			criteria.add(Restrictions.eq("patient", patient));
+		}
+		
+		if (status != null) {
+			criteria.add(Restrictions.eq("status", status));
+		}
+		
+		criteria.addOrder(Order.desc("dateCreated"));
+		
+		return criteria.list();
+	}
+	
+	/**
+	 * @see org.openmrs.module.patientqueueing.api.PatientQueueingService#savePatientQue(PatientQueue)
```

Comment:
For the parameters in the `@see` annotation, shouldn't you include the full package name instead of just the class?